File: awesome_places/lib/src/features/explore/data/models/address.dart
class Address {
  String id;
  String street;
  String city;
  String state;
  String zipCode;
  String country;

  Address({
    required this.id,
    required this.state,
    required this.street,
    required this.city,
    required this.country,
    required this.zipCode,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['id'] as String,
      state: json['state'] as String,
      street: json['street'] as String,
      city: json['city'] as String,
      country: json['country'] as String,
      zipCode: json['zipCode'] as String,
    );
  }
}

File: src/awesome_places_api/AwesomePlaces.Core/Entities/Address.cs
namespace AwesomePlaces.Core.Entities;

public class Address
{
    public Guid Id { get; protected set; } = Guid.NewGuid();
    public string Street { get; init; }
    public string City { get; init; }
    public string State { get; init; }
    public string ZipCode { get; init; }
    public string Country { get; init; }

    public Address() { }

    public Address(Guid id, string street, string city, string state, string zipCode, string country)
    {
        Id = id;
        Street = street;
        City = city;
        State = state;
        ZipCode = zipCode;
        Country = country;
    }
}